### PR TITLE
Make flash session to persist between requests. Fix #95

### LINKED ISSUE #95: Flash session is nil when doing redirect.
Hello everyone.

I'm having an issue with flash session. I'm setting the flash session and then using a redirect_to.
In the next request I'm expecting to get the value but is returning nil.

```ruby
  def call(params)
      @organization = Organization.new(params[:organization])

      if @repository.persist(@organization)
        flash[:success] = "Organization created"
        redirect_to routes.path(:manager_organizations)
      else 
        flash[:errors] = @errors = @organization.errors
      end
  end
```

In the view:
```haml
 - if flash[:success]
   = flash[:success]
```

The code is working when the flash is used in the same request. (Using the flash[:errors] in this case.)

### PR DESCRIPTION
**Pull request by donperi**
This fix issue #95

### CODE REVIEW

File: lib/lotus/action/flash.rb [hunk 1]
```diff
@@ -136,6 +145,29 @@ def remove!
...
         flash.values
       end
+
+
+      # Get the last request session flash
+      #
+      # @return [Hash] the flash of last request
+      # @since x.x.x
+      # @api private
+
```
Comment (joneslee85) [inline lib/lotus/action/flash.rb]: nitpick, please remove empty line

File: lib/lotus/action/flash.rb [hunk 2]
```diff
@@ -136,6 +145,29 @@ def remove!
...
+        flash[@last_request_id] || {}
+      end
+
+      # Store the last request_id to create the next flash with its values
+      # is current flash is not empty.
+
+      # @return [void]
+      # @since x.x.x
+      # @api private
+
```
Comment (joneslee85) [inline lib/lotus/action/flash.rb]: nitpick, please remove empty line

File: lib/lotus/action/flash.rb [hunk 3]
```diff
@@ -12,6 +12,13 @@ class Flash
       # @api private
       SESSION_KEY = :__flash
 
+      # Session key where the last request_id is stored
+      #
+      # @since x.x.x
+      # @api private
+
```
Comment (joneslee85) [inline lib/lotus/action/flash.rb]: nitpick, please remove empty line

**Comment by joneslee85** [comment]
looks legit to me

**Comment by joneslee85** [comment]
@donperi let's wait for feedback from @jodosha

**Comment by jodosha** [comment]
@donperi Thanks for this PR, it looks good. :+1: 
I'd still keep flash private API for now. It still needs to get more and more mature. /cc @joneslee85

**Pull request merged**